server/lib/rate-limit.ts

```diff
@@ -44,7 +44,7 @@ export default class RateLimit {
     if (count >= this.limit) {
       return false;
     } else {
-      cache.set(this.cacheKey, count + 1, this.expiryTimeInSeconds);
+      cache.set(this.cacheKey, count + nbCalls, this.expiryTimeInSeconds);
```

Comment:
There's an intrinsic error in this logic:
Since we're always resetting the TTL to the original `expiryTimeInSeconds`, a one-hour rate-limiter actually includes all the requests that were made within one hour of each other, which can include calls much older than one hour ago.

Reply:
(Not a blocker, I don't think we need to fix this in this PR)